docs: minor fixes to `TcpStream` API docs

vitorenesduarte opened the pull request.

carllerche commented on tokio/src/net/tcp/stream.rs:
```diff
@@ -147,14 +147,6 @@ impl TcpStream {
     /// from a future driven by a tokio runtime, otherwise runtime can be set
     /// explicitly with [`Handle::enter`](crate::runtime::Handle::enter) function.
-    ///
-    /// # Panics
```
Why is the panic block removed? I am pretty sure it can panic.

carllerche replied:
Oh, I see it is duplicated. The PR review cut off the bit above.

carllerche commented on tokio/src/net/tcp/stream.rs:
```diff
@@ -392,7 +384,7 @@ impl TcpStream {
     /// For more information about this option, see [`set_recv_buffer_size`].
     ///
-    /// [`set_recv_buffer_size`]: #tymethod.set_recv_buffer_size
+    /// [`set_recv_buffer_size`]: #method.set_recv_buffer_size
```
We probably should update links to path links instead :) I believe it would be:

```suggestion
    /// [`set_recv_buffer_size`]: TcpStream::set_recv_buffer_size
```

carllerche commented on tokio/src/net/tcp/stream.rs:
```diff
@@ -433,18 +425,9 @@ impl TcpStream {
+    /// For more information about this option, see [`set_send_buffer_size`].
     ///
-    /// [`set_keepalive`]: #tymethod.set_keepalive
+    /// [`set_send_buffer_size`]: #method.set_send_buffer_size
```
```suggestion
    /// [`set_send_buffer_size`]: TcpStream::set_send_buffer_size
```

carllerche commented on tokio/src/net/tcp/stream.rs:
```diff
@@ -488,7 +471,7 @@ impl TcpStream {
     /// For more information about this option, see [`set_keepalive`].
     ///
-    /// [`set_keepalive`]: #tymethod.set_keepalive
+    /// [`set_keepalive`]: #method.set_keepalive
```
```suggestion
    /// [`set_keepalive`]: TcpStream::set_keepalive
```

carllerche commented on tokio/src/net/tcp/stream.rs:
```diff
@@ -539,7 +522,7 @@ impl TcpStream {
     /// For more information about this option, see [`set_ttl`].
     ///
-    /// [`set_ttl`]: #tymethod.set_ttl
+    /// [`set_ttl`]: #method.set_ttl
```
```suggestion
    /// [`set_ttl`]: TcpStream::set_ttl
```

carllerche commented on tokio/src/net/tcp/stream.rs:
```diff
@@ -583,7 +566,7 @@ impl TcpStream {
     /// For more information about this option, see [`set_linger`].
     ///
-    /// [`set_linger`]: #tymethod.set_linger
+    /// [`set_linger`]: #method.set_linger
```
```suggestion
    /// [`set_linger`]: TcpStream::set_linger
```

vitorenesduarte commented:
Applied all suggestions and checked locally if links are working. The only one it's not is the one in [`TcpStream::split`](https://docs.rs/tokio/0.2.11/tokio/net/struct.TcpStream.html#method.split). I'm now remembering that I've tried to fix this before (#2078) but didn't find a way back then.

carllerche commented:
Thanks :+1: I should take a look at the `split` issue.
